File: springboot-02-ioccontainer/src/main/java/com/linkedbear/springboot/assemble/a_module/selector/WaiterDeferredImportSelector.java
package com.linkedbear.springboot.assemble.a_b_c_module_profile_conditional.selector;

import com.linkedbear.springboot.assemble.a_b_c_module_profile_conditional.component.Waiter;
import org.springframework.context.annotation.DeferredImportSelector;
import org.springframework.core.type.AnnotationMetadata;

public class WaiterDeferredImportSelector implements DeferredImportSelector {
    
    @Override
    public String[] selectImports(AnnotationMetadata importingClassMetadata) {
        System.out.println("WaiterDeferredImportSelector invoke ......");
        return new String[] {Waiter.class.getName()};
    }
}

File: springboot-02-ioccontainer/src/main/kotlin/com/linkedbear/springboot/assemble/a_module/selector/WaiterDeferredImportSelector.kt
package com.linkedbear.springboot.assemble.a_module.selector

import com.linkedbear.springboot.assemble.a_module.component.Waiter
import org.springframework.context.annotation.DeferredImportSelector
import org.springframework.core.type.AnnotationMetadata

open class WaiterDeferredImportSelector: DeferredImportSelector {
    override fun selectImports(p0: AnnotationMetadata): Array<String> {
        println("WaiterDeferredImportSelector.selectImports() invoked ......")
        return arrayOf(Waiter::class.java.name)
    }

}